crates/rpc/rpc/src/eth/builder.rs

```diff
@@ -4,52 +4,57 @@ use crate::{eth::core::EthApiInner, EthApi};
 use reth_chain_state::CanonStateSubscriptions;
 use reth_chainspec::ChainSpecProvider;
 use reth_node_api::NodePrimitives;
+use reth_rpc_eth_api::RpcNodeCore;
 use reth_rpc_eth_types::{
     fee_history::fee_history_cache_new_blocks_task, EthStateCache, EthStateCacheConfig,
     FeeHistoryCache, FeeHistoryCacheConfig, GasCap, GasPriceOracle, GasPriceOracleConfig,
 };
 use reth_rpc_server_types::constants::{
     DEFAULT_ETH_PROOF_WINDOW, DEFAULT_MAX_SIMULATE_BLOCKS, DEFAULT_PROOF_PERMITS,
 };
-use reth_storage_api::{BlockReaderIdExt, StateProviderFactory};
+use reth_storage_api::{
+    BlockReaderIdExt, ProviderBlock, ProviderHeader, ProviderReceipt, StateProviderFactory,
+};
 use reth_tasks::{pool::BlockingTaskPool, TaskSpawner, TokioTaskExecutor};
 use std::sync::Arc;
 
+// Type alias to simplify complex type
+type EthStateCacheType<N> = EthStateCache<
+    ProviderBlock<<N as RpcNodeCore>::Provider>,
+    ProviderReceipt<<N as RpcNodeCore>::Provider>,
+>;
+
 /// A helper to build the `EthApi` handler instance.
 ///
 /// This builder type contains all settings to create an [`EthApiInner`] or an [`EthApi`] instance
 /// directly.
 #[derive(Debug)]
-pub struct EthApiBuilder<Provider, Pool, Network, EvmConfig>
-where
-    Provider: BlockReaderIdExt,
-{
-    provider: Provider,
-    pool: Pool,
-    network: Network,
-    evm_config: EvmConfig,
+pub struct EthApiBuilder<N: RpcNodeCore<Provider: BlockReaderIdExt>> {
```

Comment:
```suggestion
pub struct EthApiBuilder<N: RpcNodeCore> {
```
should not need this trait bound in the type def. assuming it's required by some nested type. can tackle as sep issue.